version: '3'

services:

  dig_etl_engine:
    build: .
    volumes:
      - ./config_docker_sandbox.py:/app/dig-etl-engine/config.py
      - ${DIG_PROJECTS_DIR_PATH}:/projects_data
    ports:
      - "9999:9999"
    networks:
      - dig_net

  zookeeper:
    image: wurstmeister/zookeeper
    ports:
      - "2181:2181"
    networks:
      - dig_net

  kafka:
    image: wurstmeister/kafka
    ports:
      - "9092:9092"
    environment:
      KAFKA_ADVERTISED_HOST_NAME: ${DOCKER_HOST_NAME}
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_CREATE_TOPICS: "dig_etl_engine:1:1"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    depends_on:
      - zookeeper
    networks:
      - dig_net

  logstash:
    image: docker.elastic.co/logstash/logstash:5.5.1
    environment:
      PATH_CONFIG: /usr/share/logstash/pipeline
    volumes:
      - ./logstash/sandbox/pipeline/:/usr/share/logstash/pipeline/
      - ./logstash/sandbox/settings/logstash.yml:/usr/share/logstash/config/logstash.yml
    networks:
      - dig_net
    depends_on:
      - kafka

networks:
  dig_net:
    driver: bridge
